### ISSUE TITLE
Support for Black Box

### ISSUE THREAD

**Issue by not-a-dev-stein** [Stunkymonkey/nautilus-open-any-terminal#46]
I'd love to see support added for the new terminal that has been talked about in a couple of places, [Black Box](https://gitlab.gnome.org/raggesilver/blackbox).

I imagine it would be troublesome to add support for it tho, considering it's the first time I'm aware of a flatpak'd terminal, I'm not sure of how hard it would be to have it work.

**Comment by Stunkymonkey** [comment]
sure. could you please provide the command-line options?

**Comment by Stunkymonkey** [comment]
I did not explain it clearly what I wanted. Sorry.
Could you please provide an example on how to run `Blackbox` and be in an working-directory e.g. `~/Downloads` without any additional command.

**Comment by Stunkymonkey** [comment]
currently not supported yet. see: [BlackBox-PR-9](https://gitlab.gnome.org/raggesilver/blackbox/-/merge_requests/9)

**Comment by Aleksanaa** [comment]
> sure. could you please provide the command-line options?

Okay it is now supported with `blackbox -w ~/Downloads`

**Comment by raggesilver** [comment]
Hey @Stunkymonkey, Black Box has support specifying a working directory as of a minute ago :tada:. You can do so by passing a `--working-directory DIR` or `-w DIR` for short.

With Flatpak (the only supported distribution for Black Box), the full command is `flatpak run com.raggesilver.BlackBox --working-directory DIR`

```
Usage:
  blackbox [OPTION…] [-- COMMAND ...]

Options:
  -v, --version               Show app version
  -w, --working-directory     Set current working directory
  -c, --command               Execute command in a terminal
  -h, --help                  Show help
```

**Comment by Stunkymonkey** [comment]
thanks for the update. fixed in 1ae12e1eb3cf5d8909e3e34c0043dc6cb0e7a4be

**Comment by not-a-dev-stein** [comment]
Am I missing something or does the implementation have some issues with the Flatpak version? Installed it as user, set Black Box as the selected terminal and it doesn't work, whilst it works fine with other terminals.

**Comment by raggesilver** [comment]
Does the `-w` parameter in BlackBox work for you?

**Comment by not-a-dev-stein** [comment]
Yeah, it seems to work fine if I launch Black Box through another terminal using the -w parameter.

**Comment by Radical4ever** [comment]
> Does the `-w` parameter in BlackBox work for you?

yeah, but how do you apply this on your system so this script could work?

**Comment by Radical4ever** [comment]
now, i tried changing the parameter but it does no effect so now i guess is something related to the command used to run blackbox

**Comment by polyjitter** [comment]
Having issues too - looking at the commit, it doesn't look like it actually runs the flatpak command? It tries to run "blackbox" but that isn't provided by the flatpak version, which is the only official version of Black Box.

**Comment by Radical4ever** [comment]
maybe its worth try putting the whole command were the `--working-directory` patch is and reinstalling it to what happens

**Comment by polyjitter** [comment]
Made a pull request to fix this - #53

**Comment by polyjitter** [comment]
Was wondering if we could get a response for Black Box support?
